src/tiledb/cloud/file_ingestion/ingestion.py

```diff
@@ -0,0 +1,220 @@
+            f"""
+            ---------------------------------------------
+            - Filename: {filename}
+            - Namespace: {namespace}
+            - Destination URI: {filestore_array_uri}
+            ---------------------------------------------"""
+        )
+
+        if not tiledb.array_exists(filestore_array_uri):
+            tiledb.cloud.file.create_file(
+                namespace=namespace,
+                name=filename,
+                input_uri=file_uri,
+                output_uri=filestore_array_uri,
+                access_credentials_name=acn,
+            )
+        else:
+            logger.warning(f"Array '{filestore_array_uri}' already exists.")
+
+        if group_uri:
+            with tiledb.Group(group_uri, mode="w") as group:
+                logger.debug(f"Adding to Group {group_uri}")
+                group.add(f"tiledb://{namespace}/{filename}")
+
+
+def ingest_files(
+    dataset_uri: str,
+    source: Union[Sequence[str], str],
+    destination: str,
+    *,
+    acn: Optional[str] = None,
+    config: Optional[dict] = None,
+    namespace: Optional[str] = None,
+    include: Optional[str] = None,
+    exclude: Optional[str] = None,
+    taskgraph_name: Optional[str] = DEFAULT_FILE_INGESTION_NAME,
+    batch_size: Optional[int] = DEFAULT_BATCH_SIZE,
+    max_files: Optional[int] = None,
+    resources: Optional[Mapping[str, Any]] = None,
+    verbose: bool = False,
+) -> tiledb.cloud.dag.DAG:
+    """
+    Ingest files into a dataset.
+
+    :param dataset_uri: The dataset URI
+    :param source: URI or an iterable of URIs of input files.
+    :param destination: URI to ingest the files into.
+    :param acn: Access Credentials Name (ACN) registered in TileDB Cloud (ARN type),
+        defaults to None
+    :param config: Config dictionary, defaults to None
+    :param namespace: TileDB-Cloud namespace, defaults to None
+    :param include: UNIX shell style pattern to filter files in the search,
+        defaults to None
+    :param exclude: UNIX shell style pattern to filter files out of the search,
+        defaults to None
+    :param taskgraph_name: Optional name for taskgraph, defaults to "file-ingestion"
+    :param batch_size: Batch size for file ingestion, defaults to 100
+    :param max_files: maximum number of File URIs to read/find,
+        defaults to None (no limit)
+    :param resources: Configuration for node specs,
+        defaults to {"cpu": "1", "memory": "2Gi"}
+    :param verbose: Verbose logging, defaults to False
+    :return tiledb.cloud.dag.DAG: The resulting ingestion graph
+    """
+    # Preparation and argument cleanup
+    config = config or dict()
+    namespace = namespace or tiledb.cloud.user_profile().default_namespace_charged
+    resources = resources or DEFAULT_RESOURCES
+    dataset_uri = dataset_uri.rstrip("/")
+    logger = get_logger_wrapper(verbose)
+    if isinstance(source, str):
+        source = [source]
+
+    with tiledb.scope_ctx(config):
```

Comment:
@Shelnutt2 As I understand this, `config` should be a needed argument. Or is there a default `namespace` configuration that can be accessed by a method?

Reply:
@NikolaosPapailiou helped me understand this better. So it is resolved.